test/e2e/autonomy/autonomy_test.go

```diff
@@ -0,0 +1,330 @@
+	p "github.com/openyurtio/openyurt/test/e2e/common/pod"
+	"github.com/openyurtio/openyurt/test/e2e/util"
+	"github.com/openyurtio/openyurt/test/e2e/util/ginkgowrapper"
+	ycfg "github.com/openyurtio/openyurt/test/e2e/yurtconfig"
+)
+
+const (
+	YurtE2ENamespaceName     = "yurt-e2e-test"
+	YurtDefaultNamespaceName = "default"
+	YurtSystemNamespaceName  = "kube-system"
+	YurtCloudNodeName        = "openyurt-e2e-test-control-plane"
+	YurtEdgeNodeName         = "openyurt-e2e-test-worker"
+	YurtEdgeNode2Name        = "openyurt-e2e-test-worker2"
+	NginxServiceName         = "yurt-e2e-test-nginx"
+	CoreDNSServiceName       = "kube-dns"
+)
+
+var (
+	c                clientset.Interface
+	err              error
+	Edge2NginxPodIP  string
+	NginxServiceIP   string
+	CoreDNSServiceIP string
+
+	flannelContainerID   string
+	yurthubContainerID   string
+	kubeProxyContainerID string
+	coreDnsContainerID   string
+	coreDnsNodeName      string
+	nginxContainerID     string
+)
+
+var _ = ginkgo.Describe("edge-autonomy"+YurtE2ENamespaceName, ginkgo.Ordered, ginkgo.Label("edge-autonomy"), func() {
+	defer ginkgo.GinkgoRecover()
+	var _ = ginkgo.Describe("kubelet"+YurtE2ENamespaceName, func() {
+		gomega.RegisterFailHandler(ginkgowrapper.Fail)
+		ginkgo.It("kubelet edge-autonomy test", ginkgo.Label("edge-autonomy"), func() {
+			// restart kubelet using systemctl restart kubelet in edge nodes；check if kubelet restarted periodically
+			gomega.Eventually(func() bool {
+				_, err := exec.Command("/bin/bash", "-c", "docker exec -t openyurt-e2e-test-worker /bin/bash -c 'systemctl restart kubelet'").CombinedOutput()
+				if err != nil {
+					return false
+				}
+				return true
+			}).Should(gomega.BeTrue(), "fail to restart kubelet")
+
+			gomega.Eventually(func() bool {
+				opBytes, err := exec.Command("/bin/bash", "-c", "docker exec -t openyurt-e2e-test-worker /bin/bash -c 'curl http://127.0.0.1:10248/healthz'").CombinedOutput()
+				if err != nil {
+					return false
+				}
+				if string(opBytes) == "ok" {
+					return true
+				} else {
+					return false
+				}
+			}).WithTimeout(10*time.Second).WithPolling(1*time.Second).Should(gomega.BeTrue(), "fail to check kubelet health")
+
+			//check if nginx restarted successfully
+			gomega.Eventually(func() string {
+				opBytes, err := exec.Command("/bin/bash", "-c", "docker exec -t openyurt-e2e-test-worker /bin/bash -c 'curl http://127.0.0.1:80'").CombinedOutput()
+				if err != nil {
+					return ""
+				}
+				return string(opBytes)
+			}).Should(gomega.ContainSubstring("nginx"), "nginx pod not running")
+		})
+	})
+
+	var _ = ginkgo.Describe("flannel"+YurtE2ENamespaceName, func() {
+		ginkgo.It("flannel edge-autonomy test", ginkgo.Label("edge-autonomy"), func() {
+			// obtain flannel containerID with crictl
+			gomega.Eventually(func() bool {
+				cmd := `docker exec -t openyurt-e2e-test-worker /bin/bash -c "crictl ps | grep kube-flannel | awk '{print \$1}'"`
```

Comment:
Please use `Expect` instead of `Eventually`

Reply:
will do